extensions/resteasy-reactive/rest-client/runtime/src/main/java/io/quarkus/rest/client/reactive/CertificateUpdateEventListener.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.rest.client.reactive;
+
+import java.util.List;
+
+import jakarta.enterprise.event.Observes;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.rest.client.reactive.runtime.RestClientRecorder;
+import io.quarkus.tls.CertificateUpdatedEvent;
+import io.quarkus.tls.TlsConfiguration;
+import io.vertx.core.http.HttpClient;
+
+public class CertificateUpdateEventListener {
+
+    private static final Logger LOG = Logger.getLogger(CertificateUpdateEventListener.class);
+
+    public void onCertificateUpdate(@Observes CertificateUpdatedEvent event) {
+        String updatedTlsConfigurationName = event.name();
+        TlsConfiguration updatedTlsConfiguration = event.tlsConfiguration();
+        List<HttpClient> httpClients = RestClientRecorder.clientsUsingTlsConfig(updatedTlsConfigurationName);
+        if (httpClients.size() > 1) {
+            for (HttpClient httpClient : httpClients) {
+                httpClient.updateSSLOptions(updatedTlsConfiguration.getSSLOptions());
```

Comment:
A variant of this method receives a handler that can indicate if the update was successful or not. 

Should we log there instead? It's another inner class, but if something bad happens, we have the opportunity to log it correctly.

Reply:
Great idea, I'll do that!